lib/gooddata/models/metadata/label.rb

```diff
@@ -42,28 +42,17 @@ def find_element_value(element_id)
 
     # Gets valid elements using /validElements? API
     # @return [Array] Results
-    def get_valid_elements(url_or_params = {}, request_payload = {})
-      final_url = url_or_params
-
-      if url_or_params.is_a?(Hash)
-        default_params = {
-          limit: 1,
-          offset: 0,
-          order: 'asc'
-        }
-        params = default_params.merge(url_or_params).map { |x, v| "#{x}=#{CGI.escape(v.to_s)}" }.reduce { |acc, elem| "#{acc}&#{elem}" }
-        final_url = "#{uri}/validElements?#{params}"
-      end
-
-      results = client.post(final_url, 'validElementsRequest' => request_payload)
-
-      # Implementation of polling is based on
-      # https://opengrok.intgdc.com/source/xref/gdc-backend/src/test/java/com/gooddata/service/dao/ValidElementsDaoTest.java
-      status_url = results['uri']
-      if status_url
-        results = client.poll_on_response(status_url) do |body|
-          status = body['taskState'] && body['taskState']['status']
-          status == 'RUNNING' || status == 'PREPARED'
+    def get_valid_elements(*args)
+      results, = valid_elements(*args)
+      # TMA-775 - the validElements API can possibly return more matches than requested (usually 1)
+      # so we do a preliminary first request to check and then increase the limit if needed
+      if results['validElements']['paging']['total'].to_i != params[:limit]
+        params[:limit] = 100_000
+        results, _ = valid_elements params
```

Comment:
Style/TrailingUnderscoreVariable: Do not use trailing `_`s in parallel assignment. Prefer `results, = valid_elements params`.